modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImpl.java

```diff
@@ -612,6 +612,9 @@ public void moveMetadataToDLC(long messageId, String dlcQueueName) throws AndesE
                 .start();
         Context contextWrite = MetricManager.timer(Level.INFO, MetricsConstants.DB_WRITE).start();
 
+        //Remove the message from cache
+        removeFromCache(messageId);
```

Comment:
Started  'contextWrite' after removing messages from the cache by the PR https://github.com/wso2/andes/pull/381